pkg/services/gitopswriter/gitopswriter.go

```diff
@@ -188,109 +191,84 @@ func (dw *gitOpsDirectoryWriterSvc) RemoveApplication(ctx context.Context, app m
-	if err != nil {
-		return fmt.Errorf("failed to generate cluster automation: %w", err)
-	}
-
-	wegoConfigManifest, err := auto.GenerateWegoConfigManifest(cluster.Name, fluxNamespace, namespace)
-	if err != nil {
-		return fmt.Errorf("failed generating wego config manifest %w", err)
-	}
+type RepoWriter interface {
+	Write(ctx context.Context, repoURL gitproviders.RepoURL, branch string, manifests []gitprovider.CommitFile) error
+}
 
-	manifests := auto.Manifests()
-	manifests = append(manifests, wegoConfigManifest)
+type repoWriter struct {
+	log         logger.Logger
+	gitClient   git.Git
+	gitProvider gitproviders.GitProvider
+}
 
-	defaultBranch, err := dw.RepoWriter.GetDefaultBranch(ctx)
-	if err != nil {
-		return fmt.Errorf("failed to retrieve default branch for repository: %w", err)
+func NewRepoWriter(log logger.Logger, gitClient git.Git, gitProvider gitproviders.GitProvider) RepoWriter {
+	return &repoWriter{
+		log:         log,
+		gitClient:   gitClient,
+		gitProvider: gitProvider,
 	}
+}
 
-	// store a .keep file in the user dir
-	userKeep := automation.AutomationManifest{
-		Path:    filepath.Join(getUserDirRepoPath(cluster.Name), ".keep"),
-		Content: []byte(strconv.AppendQuote(nil, "# keep")),
+func (rw *repoWriter) Write(ctx context.Context, repoURL gitproviders.RepoURL, branch string, manifests []gitprovider.CommitFile) error {
+	// TODO: auto-merge will not work for most users
+	remover, _, err := gitrepo.CloneRepo(ctx, rw.gitClient, repoURL, branch)
+	if err != nil {
+		return fmt.Errorf("failed to clone repo: %w", err)
 	}
 
-	manifests = append(manifests, userKeep)
-
-	dw.Logger.Actionf("Associating cluster %q", cluster.Name)
-
-	if autoMerge {
-		remover, repoDir, err := dw.RepoWriter.CloneRepo(ctx, defaultBranch)
-		if err != nil {
-			return fmt.Errorf("failed to clone repo: %w", err)
-		}
-
-		defer remover()
-
-		if err := dw.RepoWriter.WriteAndMerge(ctx, repoDir, ClusterCommitMessage, manifests); err != nil {
-			return fmt.Errorf("failed writing automation to disk: %w", err)
-		}
-	} else {
-		files := []gitprovider.CommitFile{}
-
-		for _, manifest := range manifests {
-			manifestPath := manifest.Path
-			content := string(manifest.Content)
+	defer remover()
 
-			files = append(files, gitprovider.CommitFile{Path: &manifestPath, Content: &content})
+	for _, m := range manifests {
+		if err := rw.gitClient.Write(*m.Path, []byte(*m.Content)); err != nil {
+			return fmt.Errorf("failed to write manifest: %w", err)
 		}
+	}
 
-		prInfo := gitproviders.PullRequestInfo{
-			Title:         fmt.Sprintf("GitOps associate %s", cluster.Name),
-			Description:   fmt.Sprintf("Add gitops automation manifests for cluster %s", cluster.Name),
-			CommitMessage: ClusterCommitMessage,
-			TargetBranch:  defaultBranch,
-			NewBranch:     automation.GetClusterHash(cluster),
-			Files:         files,
+	err = gitrepo.CommitAndPush(ctx, rw.gitClient, ClusterCommitMessage, rw.log, func(fname string) bool {
+		for _, m := range manifests {
```

Comment:
Instead of looping through all the manifests again to filter them, could you build a map in the manifest iteration above and then lookup the filename.

Reply:
Actually, there is no need to filter. I'll just return true.